File: JavaScript/Algorithms/Recursion/factorial.js
// Write two functions that finds the factorial of any number. 
// One should use recursive, the other should just use a for loop

function findFactorialRecursive(number) {   // O(n)
  if (number === 2) {
    return 2;
  } else if (number === 1) {
    return 1
  }
  return number * findFactorialRecursive(number-1);
}

function findFactorialIterative(number) {   // O(n)
  let answer = 1;
  if (number === 2) {
    answer = 2;
  }
  for (let i = 2; i <= number; i++) {
    answer = answer * i;
  }
  return answer;
}

console.log(findFactorialRecursive(5))

File: Python/Algorithms/Recursion/factorial.py
# Write two functions that finds the factorial of any number.
# One should use recursive, the other should just use a for loop


def findFactorialRecursive(num):   # On)
    if num == 1:
        return 1
    return num * findFactorialRecursive(num-1)


def findFactorialIterative(num):   # On)
    result = 1
    for i in range(1, num+1):
        result = result * i
    return result


print(findFactorialRecursive(4))
print(findFactorialIterative)
